Tutorials/Creating-Basic-Site/Master-Template-The-Navigation-Menu/index.md

```diff
@@ -22,6 +24,34 @@ We're going to hardcode this for now. It's a good idea as you start building a s
 
 **_Save_** your changes and let's test our menu. You'll find that clicking on the Article link throws an Umbraco error as we've not created this page yet. Let's do that now.
 
+## Dynamic navigation
+
+If your navigation links are to be created from published content nodes you can loop through the child nodes.
+As an example, this has come from the default starter kit.  
+
+
+
+```html
+@inherits Umbraco.Web.Mvc.UmbracoViewPage
+@using Umbraco.Web;
+@{ 
+    var site = Model.Root();
+    var selection = site.Children.Where(x => x.IsVisible()); <!-- see below for explanation of IsVisible helper method -->
+}
+<!-- uncomment this line if you don't want the site name to appear in the top navigation -->
+<!--<a class="nav-link @Html.Raw(Model.Id == site.Id ? "navi-link--active" : "")" href="@site.Url">@site.Name</a>-->
+@foreach (var item in selection)
+{
+    <a class="nav-link @(item.IsAncestorOrSelf(Model) ? "nav-link--active" : null)" href="@item.Url">@item.Name</a>
+}
+```
+
+**The IsVisible() helper method**
```

Comment:
I think we should make this section a **Tip** instead - wouldn't that me kinda neat? 😄 

So, add lines 49 through 51 to a section looking like this:
```
:::tip
stuff here!
:::
```